gridstatus/spp.py

```diff
@@ -355,6 +354,73 @@ def get_load_forecast_mid_term(
 
         return df
 
+    def _handle_dst_floor_date(
+        self,
+        date: pd.Timestamp,
+        freq: str = "5min",
+    ) -> tuple[pd.Timestamp, bool]:
+        """Handle DST transition when flooring a date.
+
+        Args:
+            date: The date to floor
+            freq: The frequency to floor to (e.g., "5min", "h")
+
+        Returns:
+            tuple: (floored_date, is_repeated_hour)
+        """
+        is_repeated_hour = False
+        try:
+            floored_date = date.floor(freq)
+        except pytz.AmbiguousTimeError:
+            is_repeated_hour = True
+            floored_date = self.safe_for_dst_transition_floor(date, freq)
+
+        return floored_date, is_repeated_hour
+
+    def _get_short_term_forecast_data(
+        self,
+        date: str | pd.Timestamp,
+        base_url: str,
+        file_prefix: str,
+        buffer_minutes: int = 2,
+    ) -> tuple[pd.DataFrame, str] | None:
+        """Get short-term forecast data with common DST handling logic.
+
+        Args:
+            date: Date to get data for. Supports "latest" and "today"
+            base_url: Base URL for downloads
+            file_prefix: Prefix for the file name (e.g., "OP-STLF", "OP-STRF")
+            buffer_minutes: Buffer minutes for "latest" date
+
+        Returns:
+            tuple: (dataframe, url) or None if date is in the future
+        """
+        if date == "latest":
+            date = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        # Files do not exist in the future
+        if date > self.now():
+            return None
+
+        floored_date, is_repeated_hour = self._handle_dst_floor_date(date, "5min")
+
+        hour = floored_date.hour
+        padded_hour = str(hour).zfill(2)
+        padded_hour_plus_one = str((hour + 1) % 24).zfill(2)
+
+        # The 100 file does not have a "d" but 105d through 155d do.
+        add_d = is_repeated_hour and not floored_date.minute == 0
```

Comment:
Here are the load forecast and solar and wind short term forecasts during the repeated hour. Note the "d" added to file names except for the one ending 100.csv

<img width="2132" height="710" alt="Image" src="https://github.com/user-attachments/assets/f0b4ffc2-c4ab-4132-8d02-ab60cbbabdab" />
<img width="2131" height="708" alt="Image" src="https://github.com/user-attachments/assets/e32c6c9c-4184-4794-b66e-4b842a1192f3" />